src/normalize/normalizer.rs

```diff
@@ -206,22 +242,53 @@ impl Normalizer {
     where
         A: ExactSizeIterator<Item = Addr> + Clone,
     {
-        if !self.cache_maps {
-            let entries = maps::parse_filtered(pid)?;
-            self.normalize_user_addrs_impl(addrs, entries, map_files)
+        if self.procmap_query_ioctl {
+            let pid = Pid::Slf;
+            let path = format!("/proc/{pid}/maps");
+            let file = File::open(&path)
+                .with_context(|| format!("failed to open `{path}` for reading"))?;
+
+            if !self.cache_vmas {
+                let entries =
+                    move |addr| query_procmap(file.as_fd(), pid, addr, self.build_ids).transpose();
```

Comment:
heh, you always end up implementing iterators for stuff like this, so why not just do it from the get go? ;)

Reply:
Please see my other comment. It is no iterator. It's just a function.